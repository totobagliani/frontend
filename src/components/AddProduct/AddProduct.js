import React from 'react';
import styles from './styles.module.scss';

export default function AddProduct() {
  return (
    <div className="addProduct__container">
      <h2 className={styles.addproduct_header}>Add Product</h2>
      <form className="addProduct">
        <input type="text" required placeholder="Add a product Name" />
        <input type="text" required placeholder="Short Description" />
      </form>
    </div>
  );
}
