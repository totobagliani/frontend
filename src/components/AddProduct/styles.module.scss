@import '../../styles/settings.scss';

.addProduct__container {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.addproduct_header {
  font-family: $headerPageFont;
  text-align: center;
  padding: 0.5em 0;
}

.addproduct_group {
  border: none;
}

.imgform__image {
  width: 100%;
}
