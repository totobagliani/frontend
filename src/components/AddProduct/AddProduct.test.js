/* eslint-disable no-unused-vars */
/* eslint-disable comma-dangle */
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import userEvent from '@testing-library/user-event';
import AddProduct from './AddProduct';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  products: {
    products: [],
    productAdded: {
      productName: '',
      description: '',
      imageURL: '',
      isFavourite: false,
      price: 0,
      section: ''
    },
    searchResults: {
      itemsCount: 0,
      searchTerm: ''
    }
  },
  ui: {
    currentPage: 1,
    currentFilter: '',
    currentOrder: '',
    resultsVisible: false
  }
};

const store = mockStore(initialState);

let imgFile = false;

jest.mock('../../hooks/useImgData', () => ({
  ...jest.requireActual('../../hooks/useImgData'),
  handleFileChange: jest.fn().mockReturnValueOnce('algo')
}));

describe('Given the AddProduct Component', () => {
  beforeEach(() => {
    render(
      <Provider store={store}>
        <BrowserRouter>
          <AddProduct />
        </BrowserRouter>
      </Provider>
    );
    jest.clearAllMocks();
  });
  test('then must be rendered (take a snapshot)', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <AddProduct />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment(<AddProduct />)).toMatchSnapshot();
  });
  test('then when the favourite is clicked call to handleFavourite', () => {
    const favouriteCheck = screen.getByRole('checkbox');

    userEvent.click(favouriteCheck);
    expect(favouriteCheck.value).toBe('true');
  });
  test('when the press button   then fill the data-image', () => {
    const selectImageButton = screen.getByText(/Elegir Imagen/i);
    const selectImg = screen.getAllByRole('img');

    userEvent.click(selectImageButton);
    imgFile = 'algo';
    expect(selectImg).toBeTruthy();
  });
  test('when submit button', () => {
    const submitButton = screen.getByText(/Publicar/i);
    userEvent.click(submitButton);
    expect(
      screen.getByText(/Error de validacion para los siguientes campos:/i)
    ).toBeInTheDocument();
  });
});
