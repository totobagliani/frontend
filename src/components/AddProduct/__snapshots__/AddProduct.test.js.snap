// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the AddProduct Component then must be rendered (take a snapshot) 1`] = `
<DocumentFragment>
  <div
    class="addProduct__container"
  >
    <h2
      class="addproduct_header"
    >
      Añadir Producto
    </h2>
    <form
      class="addProduct"
      id="addProduct-form"
    >
      <fieldset
        class="addproduct_group"
      >
        <input
          class="addproduct__text inputtext"
          maxlength="20"
          name="productName"
          placeholder="Añade un nombre"
          type="text"
          value=""
        />
        <input
          class="addproduct__text inputtext"
          maxlength="60"
          name="description"
          placeholder="Escriba una pequeña descripción"
          type="text"
          value=""
        />
        <select
          class="addproduct__select"
          name="section"
        >
          <option
            class="addproduct__optiontitle"
            selected=""
            value=""
          >
            Sección
          </option>
          <option
            value="Baño"
          >
            Baño
          </option>
          <option
            value="Dormitorio"
          >
            Dormitorio
          </option>
          <option
            value="Cocina"
          >
            Cocina
          </option>
          <option
            value="Salon"
          >
            Salon
          </option>
          <option
            value="Terraza"
          >
            Terraza
          </option>
          <option
            value="Comedor"
          >
            Comedor
          </option>
        </select>
      </fieldset>
      <fieldset
        class="addproduct_group"
      >
        <div
          class="imgForm__container"
        >
          <img
            accept="image/png, .jpeg, .jpg, image/gif"
            alt="imagen sin cargar"
            class="imgform__image"
            src="empty.jpg"
          />
        </div>
        <input
          class="file-selector-toimage"
          form="addProduct-form"
          id="file-selector-toimage"
          name="imageURL"
          placeholder="sube una imagen"
          type="file"
        />
        <button
          class="btn btn--accept"
          form="addProduct-form"
          type="button"
        >
          Elegir Imagen
        </button>
      </fieldset>
      <fieldset
        class="addproduct_group"
      >
        <input
          class="addproduct__text inputtext"
          name="price"
          placeholder="Indica el precio "
          step=".01"
          type="number"
          value=""
        />
        <label
          class="formLabel"
          for="isFavourite"
        >
          Favorito:
          <input
            class="checkboxForm"
            name="favourite"
            type="checkbox"
            value="false"
          />
        </label>
      </fieldset>
      <div
        class="addproduct__buttons"
      >
        <button
          class="btn btn--cancel"
          type="button"
        >
          Cancelar
        </button>
        <button
          class="btn btn--accept"
          form="addProduct-form"
          type="submit"
        >
          Publicar
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
